### PR TITLE
Add feed of 50 most recent winning projects, with pagination.

### LINKED ISSUE #153: RSS feed of funded projects
We need a feed of funded projects. Since we have so many projects, this needs to be paginated in some way. I know that most RSS feeds (all RSS feeds?) don't actually make use of paginating, but this is ok. 

The main request for this page will be a regular RSS feed that people can subscribe to that shows the 50 most recently funded projects. Tacking on a ?page=x query should paginate.

Bonus points: allow output as JSON.

To keep in mind: eventually, a request to this controller for HTML (as opposed to XML) should be able to be used as a web-facing view of all of our projects (#197), but the RSS feed is the simplest version and satisfies the requirement for this issue.

Please reference this issue when submitting your pull request.

### PR DESCRIPTION
**Pull request by al3x**
Resolves #153. Extraneous changes were necessary for a working development/test environment.

### CODE REVIEW

File: app/views/home/feed.xml.builder [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    feed.entry(project) do |entry|
+      entry.title "#{project.chapter.name} – #{project.title}"
+      entry.content(project.about_project, type: 'html')
+
+      entry.author do |author|
+        author.name project.name
+      end
+
+      if project.has_images?
+        photo.link(href: project.primary_image.url, rel: 'enclosure', type: 'image/jpeg')
```
Comment (jcn) [inline app/views/home/feed.xml.builder]: This breaks because `photo` isn't actually defined yet. You probably want `project.primary_image` here.

You also don't technically need the conditional either - if you don't include it we'll just use the default [placeholder image](http://d13mwkvpspjvzo.cloudfront.net/assets/no-image-main.png) if the project doesn't have any images.

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -3,4 +3,13 @@ def index
     @chapters = Chapter.active.visitable.for_display.all
     @projects = Project.recent_winners.limit(15).all
   end
+
+  def feed
+    project_filter = ProjectFilter.new(Project.recent_winners)
+    @projects = project_filter.page(params[:page], 50).result
```
Comment (jcn) [inline app/controllers/home_controller.rb]: We're going to want some eager loading of images here, otherwise we're going to be doing 50 db calls when we go and display the images in the output.

File: app/views/home/feed.xml.builder [hunk 3]
```diff
@@ -0,0 +1,19 @@
+atom_feed do |feed|
+  feed.title 'Awesome Foundation – Winning Projects'
```
Comment (jcn) [inline app/views/home/feed.xml.builder]: We try not to include any actual text in the templates, etc. This text should go in the `config/locales/*` files.

I'd probably just also stick with "Awesome Foundation - Projects" - I don't think we tend to refer to "winners" anywhere else on the site.

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -155,7 +165,7 @@ def handle_unpublished_projects
       if current_user.logged_in?
         redirect_to chapter_project_path(current_project.chapter, current_project) and return
       else
-        render_404 and return
+        render_404 && return
```
Comment (jcn) [inline app/controllers/projects_controller.rb]: Just curious why this change was needed - I always thought the `and return` idiom was standard given how Ruby handles "and"

**Comment by jcn** [comment]
Thanks @al3x!

I left a bunch of inline comments (including a bug and an optimization).

I think I would also prefer this to go either in the `ProjectsController` class or, as was discussed in Awesome Foundation Slack, into a new `FundedProjectsController` which is going to be the home of the new public-facing project pages (the `ProjectsController` is going to remain for internal pre-funded project review).

The endpoint should probably be `/projects.xml` (if that's easy) or `/projects/feed.xml`. Ultimately, I think it'd be neat to be able to do `/chapters/nyc.xml` to get a feed of all nyc projects, so I'm thinking about developing a convention (but this is lower priority in my mind than everything else).

File: app/controllers/projects_controller.rb (hunk 4 above)
Comment (al3x) [inline app/controllers/projects_controller.rb]: Just personal habit. It's what Rubocop and most linters will recommend as well, since it's easier to just use `&&` everywhere than remember the special cases where `and` isn't problematic. 😶

Comment (jcn) [inline app/controllers/projects_controller.rb]: Sounds reasonable, though I'm not going to take the time to redo everything now.

File: app/views/home/feed.xml.builder [hunk 5]
```diff
@@ -0,0 +1,16 @@
+atom_feed do |feed|
```
Comment (jcn) [inline app/views/home/feed.xml.builder]: Since this lives under the `:locale` piece of the routes, we should probably pass `:language` in to `atom_feed` and pass in the current locale.

File: app/views/home/feed.xml.builder [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  feed.title t('feed.title')
+  feed.updated @projects[0].updated_at unless @projects.empty?
+
+  @projects.each do |project|
+    feed.entry(project) do |entry|
+      entry.title "#{project.chapter.name} – #{project.title}"
+      entry.content(project.about_project, type: 'html')
```
Comment (jcn) [inline app/views/home/feed.xml.builder]: This should be `project.funded_description` (the editable text during funding), not `about_project` (which is the original application text).

**Comment by al3x** [comment]
Okay, reorganized as requested. Rails was being difficult about `/projects.xml` and such, so for now it's just using the URL that falls out of treating `funded_projects` as a resource in the routes. Feel free to fight `406 Not Acceptable` errors if you think it's worth it.

**Comment by jcn** [comment]
@al3x I had to make a few more tweaks due to the weird sorting that was happening, but it's all fixed.

Thanks again!
